lib/trento_web/openapi/schema/checks_catalog.ex

```diff
@@ -0,0 +1,178 @@
+defmodule TrentoWeb.OpenApi.Schema.ChecksCatalog do
+  @moduledoc false
+
+  require OpenApiSpex
+  alias OpenApiSpex.Schema
+
+  alias TrentoWeb.OpenApi.Schema.Provider
+
+  defmodule Check do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "Check",
+      description: "An available check to be executed on the target infrastructure",
+      type: :object,
+      properties: %{
+        sap_system_id: %Schema{type: :string, description: "Check ID", format: :uuid},
+        name: %Schema{type: :string, description: "Check Name"},
+        description: %Schema{type: :string, description: "Check Description"},
+        remediation: %Schema{type: :string, description: "Check Remediation"},
+        implementation: %Schema{type: :string, description: "Check Implementation"},
+        labels: %Schema{type: :string, description: "Check Labels"},
+        premium: %Schema{
+          type: :boolean,
+          description: "Indicates whether the current check is a Premium check"
+        },
+        group: %Schema{
+          type: :string,
+          description: "Check Group, available when requiring a Flat Catalog"
+        },
+        provider: Provider.SupportedProviders
+      }
+    })
+  end
+
+  defmodule FlatCatalog do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "FlatCatalog",
+      description: "A flat list of the available Checks",
+      type: :array,
+      items: Check
+    })
+  end
+
+  defmodule ChecksGroup do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "ChecksGroup",
+      description: "A Group of related Checks (Corosync, Pacemaker ...)",
+      type: :object,
+      properties: %{
+        group: %Schema{type: :string, description: "Group Name"},
+        checks: FlatCatalog
+      }
+    })
+  end
+
+  defmodule ProviderCatalog do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "ProviderCatalog",
+      description: "A Provider specific Catalog, and respective values",
+      type: :object,
+      properties: %{
+        provider: %Schema{
```

Comment:
Cannot we use `Provider.SupportedProviders` here like above?

Reply:
I though about that, too.

But then I wouldn't have been able to provide a custom description for this specific case (or at least I wasn't able to find a way to do so)

```elixir
description:
    "The provider determining the values for the attached checks (azure, aws ...)",
```

I'd be fine with losing the description specificity, that'd be an acceptable tradeoff.
Using `SupportedProviders` as it is now would be a tradeoff as well I believe, because of the `:unknown` value

```elixir
defmodule SupportedProviders do
  @moduledoc false

  OpenApiSpex.schema(%{
    title: "SupportedProviders",
    type: :string,
    description: "Detected Provider where the resource is running",
    enum: [:azure, :aws, :gcp, :unknown]
  })
end
```

I think I will fix what I did
```elixir
provider: %Schema{
  title: "ChecksProvider",
  type: :string,
  description:
    "The provider determining the values for the attached checks (azure, aws ...)",
  # ??
  enum: [:azure, :aws, :gcp, :unknown]
},
```
by removing `unknown` and adding `default`.
Would that make sense?

Reply:
Ok, let's go with this

Reply:
And in fact, you did right not using `SupportedProviders` as they are different things. The catalog has some already known providers checks, while in the discovered providers, we have the `unknown` entry, as we don't detect anything outside of the big 3 cloud providers